Rubberduck.Core/UI/Command/CommandBase.cs

```diff
@@ -12,20 +12,33 @@ public abstract class CommandBase : ICommand
     {
         private static readonly List<MethodBase> ExceptionTargetSites = new List<MethodBase>();
 
-        protected CommandBase(ILogger logger)
+        protected CommandBase(ILogger logger = null)
         {
-            Logger = logger;
+            Logger = logger ?? LogManager.GetLogger(GetType().FullName);
+            CanExecuteEvaluation = (parameter => true);
         }
 
         protected ILogger Logger { get; }
-        protected virtual bool EvaluateCanExecute(object parameter) => true;
         protected abstract void OnExecute(object parameter);
 
+        protected Func<object, bool> CanExecuteEvaluation { get; private set; }
```

Comment:
This is a nitpick but the name doesn't sound right - it's asking if it can execute an evaluation, which isn't really what we mean. Maybe `EvaluateAllExecuteConditions`?

Reply:
I will change the name.